### ISSUE TITLE
Codegeneration produces the uncompiled code

### ISSUE THREAD

**Issue by ma1uta** [ngs-doo/dsl-json#97]
Project: https://github.com/ma1uta/json-example
The parent: https://github.com/ma1uta/json-example/blob/master/common/src/main/java/io/github/ma1uta/rd/json/Event.java
The first children: https://github.com/ma1uta/json-example/blob/master/common/src/main/java/io/github/ma1uta/rd/json/Fevent.java
The second children: https://github.com/ma1uta/json-example/blob/master/common/src/main/java/io/github/ma1uta/rd/json/Sevent.java

The test: https://github.com/ma1uta/json-example/blob/master/dsljson/src/test/java/io/github/ma1uta/rd/json/dsljson/DslJsonTest.java

When tried to compile the project it throws the exception: 
```
[ERROR] /home/ma1uta/dev/dsljsonexamples/common/target/generated-sources/annotations/io/github/ma1uta/rd/json/_Fevent_DslJsonConverter.java:[15,74] cannot find symbol
[ERROR]   symbol:   class C
[ERROR]   location: class io.github.ma1uta.rd.json._Fevent_DslJsonConverter.ObjectFormatConverter
[ERROR] /home/ma1uta/dev/dsljsonexamples/common/target/generated-sources/annotations/io/github/ma1uta/rd/json/_Fevent_DslJsonConverter.java:[16,75] cannot find symbol
[ERROR]   symbol:   class C
[ERROR]   location: class io.github.ma1uta.rd.json._Fevent_DslJsonConverter.ObjectFormatConverter
[ERROR] /home/ma1uta/dev/dsljsonexamples/common/target/generated-sources/annotations/io/github/ma1uta/rd/json/_Sevent_DslJsonConverter.java:[15,74] cannot find symbol
[ERROR]   symbol:   class C
[ERROR]   location: class io.github.ma1uta.rd.json._Sevent_DslJsonConverter.ObjectFormatConverter
[ERROR] /home/ma1uta/dev/dsljsonexamples/common/target/generated-sources/annotations/io/github/ma1uta/rd/json/_Sevent_DslJsonConverter.java:[16,75] cannot find symbol
[ERROR]   symbol:   class C
[ERROR]   location: class io.github.ma1uta.rd.json._Sevent_DslJsonConverter.ObjectFormatConverter

```

In short:
```java
interface Inner {};
class InnerA implements Inner {};
class InnerB implements Inner {};

@CompiledJson
abstract class X<T extends Inner> {
}

@CompiledJson
class Y extends X<InnerA> {
}

@CompiledJson
class Z extends X<InnerB> {
}
```

Won't compiled.

**Comment by zapov** [comment]
It's not compiled since you don't have any properties in those classes. It will if you add some.
But yes, it should give you some warning instead of producing invalid code.

As for the second one... I actually wanted to add a warning for that (having annotation on a place which is not analyzed).. but forgot to.
You should put annotation on the getter

**Comment by ma1uta** [comment]
Thanks for the tip, it works.

In my first example all classes have properties:
```java
public interface EventContent {
}

public class FirstContent implements EventContent {

    private String field;

    public String getField() {
        return field;
    }

    public void setField(String field) {
        this.field = field;
    }
}

@CompiledJson(deserializeDiscriminator = "type", deserializeName = Event.Type.F)
public class Fevent extends Event<FirstContent> {

    private String roomId;

    @JsonAttribute(name = "room_id")
    public String getRoomId() {
        return roomId;
    }

    public void setRoomId(String roomId) {
        this.roomId = roomId;
    }

    @Override
    public String getType() {
        return Type.F;
    }
}

@CompiledJson
public abstract class Event<C extends EventContent> {

    public static class Type {
        public static final String F = "f";

        public static final String S = "s";
    }

    private C content;

    public abstract String getType();

    public C getContent() {
        return content;
    }

    public void setContent(C content) {
        this.content = content;
    }
}
```

I added the custom converter: https://github.com/ma1uta/json-example/blob/master/dsljson/src/main/java/io/github/ma1uta/rd/json/_Fevent_DslJsonConverter.java based on generated and fix the errors.

Therefore it throws the exception:

"javax.json.bind.JsonbException: Expecting "type" attribute as first element of mixin at position: 2, following: `{"`. Found: content"

or

"java.io.IOException: Expecting "type" attribute as first element of mixin at position: 2, following: `{"`, before: `content":{"field":"d`. Found: content"

**Comment by zapov** [comment]
This setup looks like something not currently support (and nontrivial).

As for the error, what JSON are you feeding it? it says it's expecting the discriminator as first property, but it's finding the 'content' instead?

**Comment by zapov** [comment]
Great, go ahead.

I don't think you can easily add the support for discriminator not being first. That would require some memoization of parsed values up until then. I mean... technically you could do it... but it seems that it will have a lot of edge cases.
